google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -80,16 +107,110 @@ public static void main(String[] args) {
     }
   }
 
+  /**
+   * Creates a new GoogleAdsClient instance with the specified loginCustomerId.
+   *
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
+   * @return a GoogleAdsClient instance.
+   */
+  private GoogleAdsClient createGoogleAdsClient(long loginCustomerId) {
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient =
+          GoogleAdsClient.newBuilder()
+              .fromPropertiesFile()
+              .setLoginCustomerId(loginCustomerId)
+              .build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return null;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return null;
+    }
+    return googleAdsClient;
+  }
+
   /**
    * Runs the example.
    *
    * @param googleAdsClient the Google Ads API client.
    * @param managerId the root customer ID from which to begin the search.
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
    */
-  private void runExample(GoogleAdsClient googleAdsClient, long managerId) {
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId, Long loginCustomerId) {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    if (managerId != null) {
+      seedCustomerIds.add(managerId);
+    } else {
+      seedCustomerIds = getAccessibleCustomers(googleAdsClient);
+    }
+
+    Map<CustomerClient, Multimap<Long, CustomerClient>> customerClientsToHierarchies =
+        new HashMap<>();
+    List<Long> accountsWithNoInfo = new ArrayList<>();
+    // Constructs a map of account hierarchies.
+    for (long seedCustomerId : seedCustomerIds) {
+      Map<CustomerClient, Multimap<Long, CustomerClient>> customerClientToHierarchy =
+          createCustomerClientToHierarchy(loginCustomerId, seedCustomerId);
```

Comment:
The createCustomerClientToHierarchy method should actually be named as createCustomerClientsToHierarchies.
They contain many customer clients and many hierarchies.

Reply:
While the return value is technically a map of CustomerClient keys and Multimap values, it always returns a map with a single key, value pair, which is why I named it as such. I used this data structure so that we could easily merge into the map we're now calling allHierarchies once it's created. So, I think the name should remain.